src/components/Comments/CommentsBarContext.tsx

```diff
@@ -0,0 +1,93 @@
+import { useCookieState } from 'ahooks';
+import { noop } from 'lodash';
+import {
+  createContext,
+  useCallback,
+  useContext,
+  useMemo,
+  useState,
+} from 'react';
+import { ChildrenProp } from '~/common';
+
+const initialCommentsBarContext = {
+  toggleThreadComments: noop,
+  toggleCommentsBar: noop,
+  isCommentsBarOpen: false,
+  expandedThreads: [] as string[],
+  resourceCommentsTotal: 0,
+  setResourceCommentsTotal: noop,
+};
+
+export const CommentsBarContext = createContext(initialCommentsBarContext);
+
+CommentsBarContext.displayName = 'CommentsBarContext';
+
+export const CommentsBarProvider = ({ children }: ChildrenProp) => {
+  const [cookieCommentsBarOpen, setCookieCommentsBarOpen] = useCookieState(
+    'show-comments',
+    { defaultValue: 'false' }
+  );
+  const [isCommentsBarOpen, setCommentsBarOpen] = useState(
+    cookieCommentsBarOpen === 'true'
+  );
+  const [expandedThreads, setExpandedThreads] = useState<string[]>([]);
```

Comment:
I think a set would be more useful here. ahooks' as one that could be adapted.
```ts
const [currentExpandedThreads, setExpandedThreads] = useSet<string>();
const expandedThreads = useMemo(() => ({
  ...setExpandedThreads,
  has: currentExpandedThreads.has.bind(currentExpandedThreads),
  toggle: (threadId: string, next?: boolean) => {
    next = next ?? !currentExpandedThreads.has(threadId);
    setExpandedThreads[next ? 'add' : 'remove'](threadId);
  },
}), [expandedThreadsHook];
```
Then
```ts
const { expandedThreads } = useContext();

expandedThreads.has(threadId);
expandedThreads.toggle(threadId);
expandedThreads.add(threadId);
expandedThreads.remove(threadId);
```